pkg/utils/error.go

```diff
@@ -1 +1,18 @@
 package utils
+
+import (
+	"regexp"
+	"strconv"
+)
+
+func ErrorStatusCode(err error) int64 {
+	re := regexp.MustCompile(`\[(\d{3})\]`)
+	matches := re.FindStringSubmatch(err.Error())
+	var statusCode int64
+
+	if len(matches) > 1 {
+		stringCode := matches[1]
+		statusCode, _ = strconv.ParseInt(stringCode, 10, 64)
+	}
+	return statusCode
+}
```

Comment:
implement a better way to get the status code. since regex can fail in different scenarios.

Reply:
I think we only have the option to extract the status code from the error message.